### PR TITLE
feat: OOB V2

### PR DESCRIPTION
**Pull request by Baha-sk**
This change introduces OOB V2 service and client first draft.
It also injects the Accept oob property in the BDD tests.

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/client/outofbandv2/client.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	}
+
+	result := make([]oobv2.Action, len(actions))
+	copy(result, actions)
+
+	return result, nil
+}
+
+// ActionContinue allows continuing with the protocol after an action event was triggered.
+func (c *Client) ActionContinue(piID string, opts oobv2.Options) error {
```
Comment (rolsonquadras) [inline pkg/client/outofbandv2/client.go:153]: When does a client need to invoke this function ? Do they have to call `AcceptInvitation` first and then invoke this ?

Comment (Baha-sk) [reply pkg/client/outofbandv2/client.go:153]: so ActionContinue is supposed to trigger the other services, would it be the case? If it's not needed, i can definitely remove it. We just need to figure out where/how to trigger the other protocol services via oob.

**Review by Moopli — APPROVED** [review]
Selecting and triggering the followup protocol (ie, whatever the OOB message wraps) is in a followup PR, so this looks good to me :+1:

File: pkg/client/outofbandv2/client.go (hunk 1 above)
Comment (Baha-sk) [reply pkg/client/outofbandv2/client.go]: removed ActionContinue(), Actions() and ActionStop()

**Pull request merged**